File: java/amp-eventing/src/main/java/amp/eventing/streaming/EventStreamQueueItem.java
package amp.eventing.streaming;

import amp.messaging.MessageContext;
import cmf.bus.Envelope;

/**
 * Contains the elements needed to send an event in an event stream
 * User: jholmberg
 * Date: 6/3/13
 */
public class EventStreamQueueItem {

    private final MessageContext context;

    public EventStreamQueueItem(MessageContext context) {
        this.context = context;
    }

    public Envelope getEnvelope() {
        return this.context.getEnvelope();
    }

    public MessageContext getMessageContext() {
        return this.context;
    }
}

File: dotnet/source/amp.eventing/streaming/EventStreamQueueItem.cs
using amp.messaging;
using cmf.bus;

namespace amp.eventing.streaming
{
    /// <summary>
    /// Contains the elements needed to send an event in an event stream.
    /// </summary>
    public class EventStreamQueueItem
    {
        private readonly MessageContext _eventContext;

        public EventStreamQueueItem(MessageContext eventContext)
        {
            _eventContext = eventContext;
        }

        public Envelope Envelope { get { return _eventContext.Envelope; } }

        public MessageContext MessageContext { get { return _eventContext; } }
    }
}
